### PR TITLE
added new tests for validators in MVC Unit Tests Project:

### PR DESCRIPTION
**Pull request by alessandromontividiu03**
Added these new classes to test validators that were not tested before.

- ProductAttributeValueModelValidatorTests
- PredefinedProductAttributeValueModelValidatorTests
- ProductAttributeValidatorTests

### CODE REVIEW

File: src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductTagValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        public new void Setup()
+        {
+            _validator = new ProductTagValidator(_localizationService, null);
+        }
+
+        [Test]
+        public void Should_have_error_when_name_is_empty()
+        {
+            var model = new ProductTagModel();
+            model.Name = string.Empty;
```
Comment (skoshelev) [inline src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductTagValidatorTests.cs]: Please use the object initialization

File: src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductAttributeValueModelValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+using Nop.Web.Areas.Admin.Models.Catalog;
+using Nop.Web.Areas.Admin.Validators.Catalog;
+using Nop.Web.MVC.Tests.Public.Validators;
+using NUnit.Framework;
+
+namespace Nop.Web.MVC.Tests.Admin.Validators.Catalog
+{
+    public class ProductAttributeValueModelValidatorTests : BaseValidatorTests
+    {
+        private ProductAttributeValueModelValidator _validator;
```
Comment (skoshelev) [inline src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductAttributeValueModelValidatorTests.cs]: This test class must be checking max length of Name and ColorSquaresRgb filds

File: src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductTagValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using Nop.Web.Areas.Admin.Models.Catalog;
+using Nop.Web.Areas.Admin.Validators.Catalog;
+using Nop.Web.MVC.Tests.Public.Validators;
+using NUnit.Framework;
+
+namespace Nop.Web.MVC.Tests.Admin.Validators.Catalog
+{
+    [TestFixture]
+    public class ProductTagValidatorTests : BaseValidatorTests
+    {
```
Comment (skoshelev) [inline src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductTagValidatorTests.cs]: This test class must be checking max length of Name fild

File: src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+using Nop.Web.Areas.Admin.Models.Catalog;
+using Nop.Web.Areas.Admin.Validators.Catalog;
+using Nop.Web.MVC.Tests.Public.Validators;
+using NUnit.Framework;
+
+namespace Nop.Web.MVC.Tests.Admin.Validators.Catalog
+{
+    [TestFixture]
+    public class ProductValidatorTests : BaseValidatorTests
+    {
```
Comment (skoshelev) [inline src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductValidatorTests.cs]: This test class must be checking max length of Name, MetaKeywords, MetaTitle, Sku, ManufacturerPartNumber, Gtin, RequiredProductIds and AllowedQuantities filds

**Comment by skoshelev** [comment]
@alessandromontividiu03 Thank you for your request, if you making all our change we merge it

File: src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductValidatorTests.cs (hunk 4 above)
Comment (alessandromontividiu03) [reply src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductValidatorTests.cs]: sorry, but I didn't found these validations in ProductValidator class, Isn't this only to test ProductValidator class? I only found one rule for Name (NotEmpty)

Comment (skoshelev) [reply src/Tests/Nop.Web.MVC.Tests/Admin/Validators/Catalog/ProductValidatorTests.cs]: The call SetDatabaseValidationRules<Product>(dbContext) adds rules to the maximum length of the string fields

**Pull request closed without merging**